### ISSUE TITLE
Pass node public and secret keys to Service constructor

### ISSUE THREAD

**Issue by octavonce** [maidsafe/crust#966]
Currently, crust generates the node's keypair when building the Service object. This means that a node's public key will be generated each time the node is started and there is no way to pre-configure this.

A node failure would mean that the public key of the node changes making it's entry in every other node's bootstrap cache expire.

Also, why pass a UID to the service object when the public key is a good enough way to establish identity?

**Comment by povilasb** [comment]
Yep, definitely makes sense. That's actually smth we recently discussed at MaidSafe and the work is gonna start soon. See https://maidsafe.atlassian.net/browse/MAID-2730 if interested :)

**Comment by povilasb** [comment]
Also, see comments at: https://github.com/maidsafe/crust/pull/967
